core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -261,4 +272,53 @@ private Connector createConnector(ConnectionConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  private ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectionConfig namedConfig = namedConfigs.get(config.getNamedConnection());
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "No named connection named " + config.getNamedConnection());
+      }
+      return namedConfig;
+    }
+    return config;
+  }
+
+  /** Register the configuration for a named connector. */
+  public void register(String name, ConnectionConfig config) {
+    ConnectionConfig oldConfig = this.namedConfigs.putIfAbsent(name, config);
```

Comment:
What if we create the connector instead of just saving the ConnectionConfig in the Hashmap? This way we don't need `namedConfigs` and when `getConnector()` is called it returns the connector from `namedConnectors` or `connectors`.

Reply:
+1 -- I was thinking about this problem and came to the same thought. If we just create a map of connectors, we can avoid this problem of multiple concurrent hashmaps.